src/nrniv/CMakeLists.txt

```diff
@@ -238,6 +238,14 @@ set_source_files_properties(${NRN_NRNOC_SRC_DIR}/nrnversion.cpp PROPERTIES OBJEC
 set_source_files_properties(${NRN_OC_SRC_DIR}/hocusr.cpp PROPERTIES OBJECT_DEPENDS
                             ${PROJECT_BINARY_DIR}/src/oc/hocusr.h)
 
+# NVHPC/21.7 cannot compile znorm.c with -O2 or above. See also:
+# https://forums.developer.nvidia.com/t/nvc-21-7-regression-internal-compiler-error-can-only-coerce-indirect-args/184847
+if(("${CMAKE_C_COMPILER_ID}" STREQUAL "PGI" OR "${CMAKE_C_COMPILER_ID}" STREQUAL "NVHPC")
+   AND "${CMAKE_C_COMPILER_VERSION}" VERSION_GREATER_EQUAL 21.7)
+  set_source_files_properties(${PROJECT_SOURCE_DIR}/src/mesch/znorm.c PROPERTIES COMPILE_OPTIONS
+                                                                                 -Mnovect)
+endif()
```

Comment:
Just for reference, see https://github.com/neuronsimulator/nrn/pull/901 : here intel compiled was causing deadlock with AVX-512.

Just an idea : as this issue is with the vectoriser, if writing loop in different form would avoid compiler internal error then that could be an option. Then we won't need to maintain cmake patches around.